### ISSUE TITLE
installation of latest gGnome fails - because of warning given by lazy-loading of gTrack

### ISSUE THREAD

**Issue by pwaltman** [mskilab/gGnome#9]
when I tri to install gGnome (latest version as of Jan 11, 2019), I get an error as a result of a warning message that comes up when gTrack is loaded.

```
> devtools::install_github('mskilab/gGnome', force=TRUE)
Downloading GitHub repo mskilab/gGnome@master
Skipping 3 packages not available: GenomicRanges, rtracklayer, VariantAnnotation
✔  checking for file ‘/tmp/RtmpFJYpuO/remotes68c72a83fa04/mskilab-gGnome-81f661e/DESCRIPTION’ ...
─  preparing ‘gGnome’:
✔  checking DESCRIPTION meta-information ...
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
   Removed empty directory ‘gGnome/inst/extdata/gTrack.js’
─  building ‘gGnome_0.1.tar.gz’
   
Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)
* installing *source* package ‘gGnome’ ...
** R
** inst
** byte-compile and prepare package for lazy loading
Error: package or namespace load failed for ‘gTrack’:
 (converted from warning) multiple methods tables found for ‘seqinfo<-’
Error : package ‘gTrack’ could not be loaded
ERROR: lazy loading failed for package ‘gGnome’
* removing ‘/usr/local/lib/R/site-library/gGnome’
* restoring previous ‘/usr/local/lib/R/site-library/gGnome’
Error in i.p(...) : 
  (converted from warning) installation of package ‘/tmp/RtmpFJYpuO/file68c72afe8e0f/gGnome_0.1.tar.gz’ had non-zero exit status
```

**Comment by imielinski** [comment]
Warnings should not prevent installation - did you change options(warn=) in your R sesh




On Sat, Jan 12, 2019 at 1:02 PM -0500, "Peter Waltman" <notifications@github.com<mailto:notifications@github.com>> wrote:


when I tri to install gGnome (latest version as of Jan 11, 2019), I get an error as a result of a warning message that comes up when gTrack is loaded.

> devtools::install_github('mskilab/gGnome', force=TRUE)
Downloading GitHub repo mskilab/gGnome@master
Skipping 3 packages not available: GenomicRanges, rtracklayer, VariantAnnotation
✔  checking for file ‘/tmp/RtmpFJYpuO/remotes68c72a83fa04/mskilab-gGnome-81f661e/DESCRIPTION’ ...
─  preparing ‘gGnome’:
✔  checking DESCRIPTION meta-information ...
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
   Removed empty directory ‘gGnome/inst/extdata/gTrack.js’
─  building ‘gGnome_0.1.tar.gz’

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)
* installing *source* package ‘gGnome’ ...
** R
** inst
** byte-compile and prepare package for lazy loading
Error: package or namespace load failed for ‘gTrack’:
 (converted from warning) multiple methods tables found for ‘seqinfo<-’
Error : package ‘gTrack’ could not be loaded
ERROR: lazy loading failed for package ‘gGnome’
* removing ‘/usr/local/lib/R/site-library/gGnome’
* restoring previous ‘/usr/local/lib/R/site-library/gGnome’
Error in i.p(...) :
  (converted from warning) installation of package ‘/tmp/RtmpFJYpuO/file68c72afe8e0f/gGnome_0.1.tar.gz’ had non-zero exit status


—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_mskilab_gGnome_issues_9&d=DwMFaQ&c=C9X8xNkG_lwP_-eFHTGejw&r=keZBxH6JuK9p4-nb6vzTlgDQy-idI21QN_zgWo2_gR0&m=N3saOsV0XUNT1IvHr7Dclb3hItEAfJhs2pWqomXPOls&s=AZTa7mqetkqvaFqXEwk062lwxshU_0Pk_p9q-vrVA_w&e=>, or mute the thread<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AEtC47wLP6rZL2iuTVWoJsQdqSd1lLlrks5vCiMqgaJpZM4Z8xde&d=DwMFaQ&c=C9X8xNkG_lwP_-eFHTGejw&r=keZBxH6JuK9p4-nb6vzTlgDQy-idI21QN_zgWo2_gR0&m=N3saOsV0XUNT1IvHr7Dclb3hItEAfJhs2pWqomXPOls&s=o_MRdjaiL9v3PAO3jROR6F9kCGR8Xf536_s9aaReMA8&e=>.

________________________________
This message is for the recipient’s use only, and may contain confidential, privileged or protected information. Any unauthorized use or dissemination of this communication is prohibited. If you received this message in error, please immediately notify the sender and destroy all copies of this message. The recipient should check this email and any attachments for the presence of viruses, as we accept no liability for any damage caused by any virus transmitted by this email.

**Comment by pwaltman** [comment]
Yeah, I had that thought too, but when I first got the error, I set options( warn=-1), and even tried to suppress the warning by callining install_github within suppressWarnings, i.e. 
`suppressWarnings( install_github() )`

but still got the error.

After a little digging, it appears that treating warnings as errors is the default behavior for the travis tool/build utility: https://docs.travis-ci.com/user/languages/r/#package-check-options. In theory, it's something that you can disable that by adding a parameter or setting in the travis.yaml file.

**Comment by imielinski** [comment]
They both pass Travis, so not sure why you’re getting an error.  I was also able to install both gGnome and gTrack on a new mac via install_github

Maybe just try again with a clean R installation or R library directory?

**Comment by pwaltman** [comment]
That's interesting. What version of R are you testing on? I was able to install it on our hpc, where I'm running a self-compiled version of R, which is version 3.5.1.  Additionally, I tried to install it on another clean machine (a 3rd machine) where the install failed again, for the same reason.

In both cases, where the installs failed, the machines are running Ubuntu 18.04, and have R version 3.5.2, provided by CRAN (installed from .debs). In theory, I could try to compile my own version of R v.3.5.2 to see if it's an issue with the version or if there is something specific to the version provided by the repo, but I won't be able to do that for a couple of days.

**Comment by pwaltman** [comment]
As a follow-up, I think the problem is a versioning issue with R. To be specific, I was able to install gGnome using R versions 3.5.0 and 3.5.1, both of which I compiled and installed manually. However, when I tried to install it to an instance of R version 3.5.2, which I also manually compiled and installed, I got the same error that I reported. So, it would seem that they've made some sort of change with version 3.5.2 that prevents the installer from ignoring warning messages during the install.
